lib/utils.ex

```diff
@@ -0,0 +1,26 @@
+defmodule HashRing.Utils do
+  @compile :native
+
+  @type t :: __MODULE__
+
+  def hash(key) when is_binary(key) do
+    <<_ :: binary-size(8), value :: unsigned-little-integer-size(64)>> = :erlang.md5(key)
+    value
+  end
+  def hash(key), do: hash("#{key}")
+
+  def gen_items([], _num_replicas), do: {}
+  def gen_items(nodes, num_replicas) do
+    gen_items(nodes, Enum.to_list(0..(num_replicas - 1)), [])
+  end
+
+  def gen_items([], _replicas, items) do
+    items
+      |> Enum.sort(&(elem(&1, 0) < elem(&2, 0)))
```

Comment:
isn't this the default behavior of sort? 
IOW `Enum.sort(items)`

Reply:
default behavior of sort is erlang term sorting specification, which will sort by the other items in the tuple too.

Reply:
Are you concerned about the speed? Because you shouldn't be. 
I benchmarked this using benchfella. Sorting a million element shuffled list of a tuple of ten random integers, you get this:
```elixir
## SortingBench
benchmark name             iterations   average time
Enum.sort(elems)                    1   1555209.00 µs/op
Enum.sort(elems, sort_fn)           1   2808098.00 µs/op
```
The custom sorter is significantly slower. It gets worse if the first element is a string or an atom. 
And it's harder to read.

You only gain benefit if every single tuple has the same element in the first position, if there are even two choices, (basically modding the first element by 2) the default sorter is faster.